metrics-server/src/lib.rs

```diff
@@ -145,7 +145,6 @@ pub fn start_metrics_server<TNetwork: Network>(
         let task_monitors = task_monitors.to_vec();
         spawn({
             let task_metrics = Arc::clone(&task_metrics);
-            let task_monitors = task_monitors;
```

Comment:
I think #3165 has the better fix for this code. Your change silences the warning, but the code around it looks really weird.